libs/labelbox/src/labelbox/adv_client.py

```diff
@@ -0,0 +1,115 @@
+import io
+import json
+import logging
+from typing import Dict, Any, Optional, List, Callable
+from urllib.parse import urlparse
+
+import requests
+from requests import Session, Response
+
+logger = logging.getLogger(__name__)
+
+
+class AdvClient:
+
+    def __init__(self, endpoint: str, api_key: str):
+        self.endpoint = endpoint
+        self.api_key = api_key
+        self.session = self._create_session()
+
+    def create_embedding(self, name: str, dims: int) -> Dict[str, Any]:
+        data = {"name": name, "dims": dims}
+        return self._request("POST", "/adv/v1/embeddings", data)
+
+    def delete_embedding(self, id: str):
+        return self._request("DELETE", f"/adv/v1/embeddings/{id}")
+
+    def get_embedding(self, id: str) -> Dict[str, Any]:
+        return self._request("GET", f"/adv/v1/embeddings/{id}")
+
+    def get_embeddings(self) -> List[Dict[str, Any]]:
+        return self._request("GET", "/adv/v1/embeddings").get("results", [])
+
+    def import_vectors_from_file(self, id: str, file_path: str, callback=None):
```

Comment:
if there is no callback, is there way for a user to know result / errors or success. / failure at least?

Reply:
I essentially copied this method verbatim from the existing `advlib`, I agree it's not too user-friendly but it's how it currently works and I decided to leave as-is.

Perhaps we leave this as a follow-up item if/when there are complaints but TBH I'm not sure many customers use this feature so might not be worth the effort for now.

To answer your question, the expectation might be that the user uses `get_imported_vector_count` to validate the (async) operation has completed.